src/Agent.Plugins/PipelineCache/TarUtils.cs

```diff
@@ -76,10 +73,8 @@ public static class TarUtils
 
             Directory.CreateDirectory(targetDirectory);
             
-            DedupIdentifier dedupId = DedupIdentifier.Create(manifest.Items.Single(i => i.Path == $"/{archiveFileName}").Blob.Id);
-            bool does7zExists = isWindows ? CheckIf7ZExists() : false;
-            string processFileName = (does7zExists) ? "7z" : "tar";
-            string processArguments = (does7zExists) ? $"x -si -aoa -o{targetDirectory} -ttar" : $"-xf - -C {targetDirectory}";
+            DedupIdentifier dedupId = DedupIdentifier.Create(manifest.Items.Single(i => i.Path.EndsWith(archive, StringComparison.Ordinal)).Blob.Id);
```

Comment:
>Ordinal [](start = 130, length = 7)

StringComparison.OrdinalIgnoreCase

Reply:
Meant for OrdinalIgnoreCase to begin with.